framework/framework.go

```diff
@@ -94,15 +101,20 @@ func New(config Config) (*Framework, error) {
 		return nil, microerror.Maskf(invalidConfigError, "config.ResourceRouter must not be empty")
 	}
 
+	// Settings.
+	if config.BackOffFactory == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.BackOffFactory must not be empty")
+	}
+	if config.InitCtxFunc == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.InitCtxFunc must not be empty")
+	}
```

Comment:
Although those are functions to call I can imagine replacing them with few constants to produce values created by these functions. Also they have defaults now. So they are settings?

Reply:
I am not sure about this. When you think it is a good thing lets move forward.

Reply:
I'm uncertain as well, but e.g. Caller in micrologger is a setting.

Reply:
Motivation here is to be able to change that in all operators globally. Calling it settings is lesser issue for me.